src/main/kotlin/ai/blindspot/ktoolz/extensions/OtherExtensions.kt

```diff
@@ -95,3 +71,19 @@ fun <A : Any, B : Any> Pair<A?, B?>.propagateNull(): Pair<A, B>? {
  * is not allowed.
  */
 inline fun <reified T : Any> kClass(): KClass<T> = T::class
+
+/**
+ * Calls the specified function [block] with `this` value as its receiver if and only if the [shouldApplyBlock] lambda returns true.
+ * Returns `this` value.
+ */
+inline fun <T : Any> T.applyIf(shouldApplyBlock: () -> Boolean, block: T.() -> Unit): T = applyIf(shouldApplyBlock(), block)
```

Comment:
I think that you should submit `T` into the `shouldApplyBlock`. Either as this or lambda parameter